Formula/git.rb

```diff
@@ -85,12 +96,14 @@ def install
     git_core = libexec/"git-core"
 
     # Install the macOS keychain credential helper
-    cd "contrib/credential/osxkeychain" do
-      system "make", "CC=#{ENV.cc}",
-                     "CFLAGS=#{ENV.cflags}",
-                     "LDFLAGS=#{ENV.ldflags}"
-      git_core.install "git-credential-osxkeychain"
-      system "make", "clean"
+    on_macs do
```

Comment:
```suggestion
    on_macos do
```
Let's run this through CI in case there are other subtle errors I missed 🙂